gcloud-java-compute/src/main/java/com/google/cloud/compute/Compute.java

```diff
@@ -0,0 +1,2709 @@
+     * {@code value} is interpreted as a regular expression using RE2 syntax. {@code value} must
+     * match the entire field.
+     *
+     * @see <a href="https://github.com/google/re2/wiki/Syntax">RE2</a>
+     */
+    public static NetworkFilter equals(NetworkField field, String value) {
+      return new NetworkFilter(checkNotNull(field), ComparisonOperator.EQ, checkNotNull(value));
+    }
+
+    /**
+     * Returns a not-equals filter for the given field and string value. For string fields,
+     * {@code value} is interpreted as a regular expression using RE2 syntax. {@code value} must
+     * match the entire field.
+     *
+     * @see <a href="https://github.com/google/re2/wiki/Syntax">RE2</a>
+     */
+    public static NetworkFilter notEquals(NetworkField field, String value) {
+      return new NetworkFilter(checkNotNull(field), ComparisonOperator.NE, checkNotNull(value));
+    }
+
+    /**
+     * Returns a equals filter for the given field and boolean value.
+     */
+    public static NetworkFilter equals(NetworkField field, boolean value) {
+      return new NetworkFilter(checkNotNull(field), ComparisonOperator.EQ, value);
+    }
+
+    /**
+     * Returns a not-equals filter for the given field and boolean value.
+     */
+    public static NetworkFilter notEquals(NetworkField field, boolean value) {
+      return new NetworkFilter(checkNotNull(field), ComparisonOperator.NE, value);
+    }
+  }
+
+  /**
+   * Class for filtering instance lists.
+   */
+  class InstanceFilter extends ListFilter {
+
+    private static final long serialVersionUID = 679008888882025686L;
+
+    private InstanceFilter(InstanceField field, ComparisonOperator operator, Object value) {
+      super(field.selector(), operator, value);
+    }
+
+    /**
+     * Returns an equals filter for the given field and string value. For string fields,
+     * {@code value} is interpreted as a regular expression using RE2 syntax. {@code value} must
+     * match the entire field.
+     *
+     * @see <a href="https://github.com/google/re2/wiki/Syntax">RE2</a>
+     */
+    public static InstanceFilter equals(InstanceField field, String value) {
+      return new InstanceFilter(checkNotNull(field), ComparisonOperator.EQ, checkNotNull(value));
+    }
+
+    /**
+     * Returns a not-equals filter for the given field and string value. For string fields,
+     * {@code value} is interpreted as a regular expression using RE2 syntax. {@code value} must
+     * match the entire field.
+     *
+     * @see <a href="https://github.com/google/re2/wiki/Syntax">RE2</a>
+     */
+    public static InstanceFilter notEquals(InstanceField field, String value) {
+      return new InstanceFilter(checkNotNull(field), ComparisonOperator.NE, checkNotNull(value));
+    }
+
+    /**
+     * Returns a equals filter for the given field and boolean value.
+     */
+    public static InstanceFilter equals(InstanceField field, boolean value) {
```

Comment:
![Codacy](https://www.codacy.com/assets/images/favicon.png) Issue found: [The method name and parameter number are suspiciously close to equals(Object)](https://www.codacy.com/app/mziccard/gcloud-java/file/2740139423/issues/source?bid=3123104&fileBranchId=3354337#l1050)